packages/admin/admin/src/common/buttons/feedback/FeedbackButton.tsx

```diff
@@ -0,0 +1,166 @@
+        classes,
+        disabled,
+        startIcon,
+        endIcon,
+        tooltipSuccessMessage,
+        tooltipErrorMessage,
+        slotProps,
+
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminFeedbackButton",
+    });
+
+    const [displayState, setDisplayState] = React.useState<FeedbackButtonDisplayState>("idle");
+
+    const ownerState: OwnerState = {
+        displayState: displayState,
+    };
+
+    const resolveTooltipForDisplayState = (displayState: FeedbackButtonDisplayState) => {
+        if (displayState === "success") {
+            return "success";
+        } else if (displayState === "error") {
+            return "error";
+        } else {
+            return "neutral";
+        }
+    };
+
+    React.useEffect(() => {
+        let timeoutId: number | undefined;
+        let timeoutDuration: number | undefined;
+        let newDisplayState: FeedbackButtonDisplayState;
+
+        if (displayState === "idle" && loading) {
+            setDisplayState("loading");
+        } else if (displayState === "loading" && hasErrors) {
+            timeoutDuration = 500;
+            newDisplayState = "loading";
+        } else if (displayState === "loading" && !loading && !hasErrors) {
+            timeoutDuration = 500;
+            newDisplayState = "success";
+        } else if (displayState === "error") {
+            timeoutDuration = 5000;
+            newDisplayState = "idle";
+        } else if (displayState === "success") {
+            timeoutDuration = 2000;
+            newDisplayState = "idle";
+        }
+
+        if (timeoutDuration !== undefined) {
+            timeoutId = window.setTimeout(() => {
+                setDisplayState(newDisplayState);
+            }, timeoutDuration);
+        }
+        return () => {
+            if (timeoutId !== undefined) {
+                window.clearTimeout(timeoutId);
+            }
+        };
+    }, [displayState, loading, hasErrors]);
+
+    const tooltip = (
+        <Tooltip
+            title={displayState === "error" ? tooltipErrorMessage : tooltipSuccessMessage}
+            open={displayState === "error" || displayState === "success"}
+            placement={endIcon && !startIcon ? "top-end" : "top-start"}
+            variant={resolveTooltipForDisplayState(displayState)}
+            {...slotProps?.tooltip}
+        >
+            <span>{startIcon ? startIcon : endIcon}</span>
+        </Tooltip>
+    );
+
+    return (
+        <Root
+            ownerState={ownerState}
+            loading={loading}
+            {...slotProps}
+            {...restProps}
+            variant={variant}
+            color={color}
+            disabled={disabled || displayState === "loading"}
+            loadingPosition={startIcon ? "start" : "end"}
+            loadingIndicator={<ThreeDotSaving />}
+            startIcon={startIcon && tooltip}
+            endIcon={endIcon && !startIcon ? tooltip : <span>{endIcon}</span>}
```

Comment:
As far as I understood, it is only possible to use custom components in the tooltip, if there is the possibility to pass refs to the underlying element. Otherwise, the tooltip is not displayed. The span avoids this. 

https://mui.com/material-ui/react-tooltip/#custom-child-element

Reply:
Yes, but this is not inside the tooltip.